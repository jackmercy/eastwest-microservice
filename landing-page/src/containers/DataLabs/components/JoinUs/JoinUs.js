import Link from 'next/link';
import { memo } from 'react';
import { useIntl } from 'react-intl';
import PropTypes from 'prop-types';

import { Button } from '../../../../components';
import { Content } from '../../../components';

import useStyles from './joinUs.styles';

const JoinUs = ({ description, defaultReason }) => {
  const intl = useIntl();
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Content className={classes.content}>
        <h2 className={classes.description} data-test="join-us-description">
          {description}
        </h2>
        <Link
          href={`/contact/?defaultReason=${defaultReason}#contact-form`}
          passHref
          as="/contact/#contact-form"
        >
          <Button
            tag="a"
            color="secondary"
            size="large"
            keepSizeOnMobile
            className={classes.button}
            dataTest="join-us-cta"
          >
            {intl.formatMessage({ id: 'DATA_LABS.JOIN_US.BUTTON' })}
          </Button>
        </Link>
      </Content>
    </div>
  );
};
JoinUs.propTypes = {
  description: PropTypes.string.isRequired,
  defaultReason: PropTypes.number.isRequired,
};

export default memo(JoinUs);
