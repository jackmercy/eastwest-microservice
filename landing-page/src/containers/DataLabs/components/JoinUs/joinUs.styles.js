import { createUseStyles } from 'react-jss';

import { rem } from '../../../../utils/jss';
import COLORS from '../../../../constants/COLORS';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    width: '100%',
    height: rem(240),
    backgroundColor: COLORS.primary,
    backgroundImage: 'url(/static/images/data-labs/join-us.png)',
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'right',
  },
  content: { display: 'flex', alignItems: 'center', height: '100%' },
  description: {
    margin: 0,
    fontWeight: 900,
    fontSize: rem(56),
    lineHeight: rem(72),
    color: COLORS.textWhite,
  },
  button: { flex: 'none', marginLeft: rem(45) },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: {
      height: rem(280),
      backgroundImage: 'url(/static/images/data-labs/join-us-mobile.png)',
      backgroundPosition: 'left',
    },
    content: { flexDirection: 'column', justifyContent: 'center' },
    description: {
      width: '100%',
      fontSize: rem(32),
      lineHeight: rem(48),
      textAlign: 'center',
    },
    button: {
      padding: [0, rem(32)],
      marginLeft: 0,
      marginTop: rem(24),
      boxShadow: '2px 4px 12px rgba(0, 0, 0, 0.15)',
    },
  },
});
