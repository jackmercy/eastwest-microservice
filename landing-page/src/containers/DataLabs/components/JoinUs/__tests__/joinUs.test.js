import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import JoinUs from '../JoinUs';

describe('Testing DataLabs components > JoinUs', () => {
  let wrapper;
  const props = {
    description: 'Something',
    defaultReason: 4,
  };

  beforeAll(() => {
    wrapper = mount(<JoinUs {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<JoinUs {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a description', () => {
    expect(wrapper.find('[data-test="join-us-description"]')).toHaveLength(1);
  });

  it('should have a button get started', () => {
    expect(wrapper.find('[data-test="join-us-cta"]')).toHaveLength(1);
  });
});
