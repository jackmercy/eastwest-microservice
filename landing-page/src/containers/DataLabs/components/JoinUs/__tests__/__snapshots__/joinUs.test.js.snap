// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DataLabs components > JoinUs should render match snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <Memo(Content)
    className="content-0-2-2"
  >
    <h2
      className="description-0-2-3"
      data-test="join-us-description"
    >
      Something
    </h2>
    <Link
      as="/contact/#contact-form"
      href="/contact/?defaultReason=4#contact-form"
      passHref={true}
    >
      <Memo([object Object])
        className="button-0-2-4"
        color="secondary"
        dataTest="join-us-cta"
        keepSizeOnMobile={true}
        size="large"
        tag="a"
      >
        Join now!
      </Memo([object Object])>
    </Link>
  </Memo(Content)>
</div>
`;
